### ISSUE TITLE
MD054: Link text that uses array syntax is flagged as `shortcut` link style

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#488]
### rumdl version

0.1.39

### Installation method

mise

### Bug description

MD054 (https://github.com/rvben/rumdl/blob/7f7f5a89a9d45fe3219f41aa2cf396eedcc61c1b/docs/md054.md) allows projects to establish a consistent link style.

However, the rule can catch instances of array syntax used in backticks that looks like like `shortcut` link syntax (https://github.com/rvben/rumdl/blob/7f7f5a89a9d45fe3219f41aa2cf396eedcc61c1b/docs/md054.md#link-and-image-styles).

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   ---
   title: heading
   ---
   
   Link to information about [`[myArray]`](#information-on-myarray).
   
   ## Information on `[myArray]`
   
   Some section content.
   ```
1. Create a `.rumdl.toml` file with this content:
   ```toml
   [MD054]
   shortcut = false
   ```
1. Run `rumdl check` on the Markdown file.

### Expected behavior

In this scenario, I'd expect no problems to be flagged.

### Actual behavior

`rumdl` outputs the following:

```shell
test.md:5:27: [MD054] Link/image style 'shortcut' is not allowed
```

### Configuration

```toml
[MD054]
shortcut = false
```

### Operating system

macOS

**Comment by eread** [comment]
I should also mention that this line is also flagged:

```markdown
Link to information about [`myarray[]`](#information-on-myarray).
```

So I think lots of ways of expressing "this is an array" using backticks would be caught.

**Comment by rvben** [comment]
Fixed in [v0.1.41](https://github.com/rvben/rumdl/releases/tag/v0.1.41). Bracket text like `[0]`, `[i]`, and `[key]` without a matching link definition is no longer flagged as a shortcut link. Thanks for reporting @eread!

**Comment by eread** [comment]
> Fixed in [v0.1.41](https://github.com/rvben/rumdl/releases/tag/v0.1.41). Bracket text like `[0]`, `[i]`, and `[key]` without a matching link definition is no longer flagged as a shortcut link. Thanks for reporting [@eread](https://github.com/eread)!

@rvben Fixed!
